src/nupic/algorithms/Anomaly.cpp

```diff
@@ -0,0 +1,136 @@
+/* ---------------------------------------------------------------------
+ * Numenta Platform for Intelligent Computing (NuPIC)
+ * Copyright (C) 2016, Numenta, Inc.  Unless you have an agreement
+ * with Numenta, Inc., for a separate license for this software code, the
+ * following terms and conditions apply:
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses.
+ *
+ * http://numenta.org/licenses/
+ * ---------------------------------------------------------------------
+ */
+
+
+#include <vector>
+#include <numeric>
+#include <algorithm>
+#include "nupic/algorithms/Anomaly.hpp"
+
+using namespace std;
+using namespace nupic::algorithms::anomaly;
+
+/**
+Computes the raw anomaly score.
+
+The raw anomaly score is the fraction of active columns not predicted.
+
+@param activeColumns: array of active column indices
+@param prevPredictedColumns: array of columns indices predicted in prev step
+@return anomaly score 0..1 (float)
+*/
+float computeRawAnomalyScore(const vector<int>& active, const vector<int>& predicted)
+{
+  // Return 0 if no active columns are present
+  if (active.size() == 0) {
+    return 0.0f;
+  }
+	
+  // Count active columns that were predicted
+  int score = accumulate(active.begin(), active.end(), 0,
```

Comment:
This could be a little more efficient with one of these methods:

1. Sort both vectors and then iterate through them, advancing whichever iterator has a smaller value and advancing count if they are equal.
2. Dump active cells into a set and then count each value in predicted that exists in the set.

Both of these are O(n logn) where the original is ~O(n^2).